cryptoki/src/session/mod.rs

```diff
@@ -0,0 +1,478 @@
+        slot_token_management::init_pin(self, pin)
+    }
+
+    /// Changes the PIN of either the currently logged in user or of the `CKU_USER` if no user is
+    /// logged in.
+    pub fn set_pin(&self, old_pin: &str, new_pin: &str) -> Result<()> {
+        slot_token_management::set_pin(self, old_pin, new_pin)
+    }
+
+    /// Close a session
+    /// This will be called on drop as well.
+    pub fn close(self) {}
+
+    /// Log a session in.
+    ///
+    /// # Arguments
+    ///
+    /// * `user_type` - The type of user to log in as
+    /// * `pin` - The PIN to use, or `None` if you wish to use the protected authentication path
+    ///
+    /// _NOTE: By passing `None` into `login`, you must ensure that the
+    /// [CKF_PROTECTED_AUTHENTICATION_PATH] flag is set in the `TokenFlags`._
+    pub fn login(&self, user_type: UserType, pin: Option<&str>) -> Result<()> {
+        session_management::login(self, user_type, pin)
+    }
+
+    /// Log a session out
+    pub fn logout(&self) -> Result<()> {
+        session_management::logout(self)
+    }
+
+    /// Returns the information about a session
+    pub fn get_session_info(&self) -> Result<SessionInfo> {
+        session_management::get_session_info(self)
+    }
+
+    /// Search for session objects matching a template
+    pub fn find_objects(&self, template: &[Attribute]) -> Result<Vec<ObjectHandle>> {
+        object_management::find_objects(self, template)
+    }
+
+    /// Create a new object
+    pub fn create_object(&self, template: &[Attribute]) -> Result<ObjectHandle> {
+        object_management::create_object(self, template)
+    }
+
+    /// Destroy an object
+    pub fn destroy_object(&self, object: ObjectHandle) -> Result<()> {
+        object_management::destroy_object(self, object)
+    }
+
+    /// Get the attribute info of an object: if the attribute is present and its size.
+    ///
+    /// # Arguments
+    ///
+    /// * `object` - The [ObjectHandle] used to reference the object
+    /// * `attributes` - The list of attributes to get the information of
+    ///
+    /// # Returns
+    ///
+    /// This function will return a Vector of [AttributeInfo] enums that will either contain
+    /// the size of the requested attribute, [AttributeInfo::TypeInvalid] if the attribute is not a
+    /// valid type for the object, or [AttributeInfo::Sensitive] if the requested attribute is
+    /// sensitive and will not be returned to the user.
+    ///
+    /// The list of returned attributes is 1-to-1 matched with the provided vector of attribute
+    /// types.  If you wish, you may create a hash table simply by:
+    ///
+    /// ```rust
+    /// use cryptoki::Pkcs11;
+    /// use cryptoki::types::locking::CInitializeArgs;
+    /// use cryptoki::types::object::AttributeType;
+    /// use cryptoki::types::session::UserType;
+    /// use cryptoki::types::SessionFlags;
+    /// use std::collections::HashMap;
+    /// use std::env;
```

Comment:
I think because of your changes to the layout of the crate, these `use` statements are failing